lib/iris/_shapefiles.py

```diff
@@ -66,52 +74,64 @@ def create_shapefile_mask(
             sgeom.MultiLineString,
         ),
     ):
-        weights = False
+        minimum_weight = 0.0
         warnings.warn(
-            "Shape is of invalid type for minimum weight masking, must use a Polygon rather than Line shape.\n Masking based off intersection instead. ",
-            IrisUserWarning,
+            """Shape is of invalid type for minimum weight masking,
+            must use a Polygon rather than Line shape.\n
+              Masking based off intersection instead. """,
+            IrisDefaultingWarning,
         )
-    else:
-        weights = True
 
     # prepare shape
-    trans_geo = _transform_coord_system(geometry, cube, shape_coord_system)
-    prepped.prep(trans_geo)
-
+    trans_geo = _transform_coord_system(geometry, cube)
     # prepare 2D cube
     y_name, x_name = _cube_primary_xy_coord_names(cube)
     cube_2d = cube.slices([y_name, x_name]).next()
-    _cube_xy_guessbounds(cube_2d)
-    xmod = cube.coord(x_name).units.modulus
-    ymod = cube.coord(y_name).units.modulus
-    mask_template = np.zeros(cube_2d.shape, dtype=np.float64)
+    for coord in cube_2d.dim_coords:
+        if not coord.has_bounds():
+            coord.guess_bounds()
+
+    y_coord, x_coord = [cube_2d.coord(n) for n in (y_name, x_name)]
+    x_bounds = _get_mod_rebased_coord_bounds(x_coord)
+    y_bounds = _get_mod_rebased_coord_bounds(y_coord)
+    # prepare array for dark
+    bounds_array = np.asarray(list(product(x_bounds, y_bounds)))
+    box_template = _template_func(bounds_array)
+    # shapely can do lazy evaluation of intersections if it's given a list of grid box shapes
+    # delayed lets us do it in parallel
+    intersect_template = dask.delayed(
+        shapely.intersects(trans_geo, box_template)
+    )
```

Comment:
Most of the improvement comes from giving `shapely.intersects` a interable, but I have a custom large file test for checking the improvement of dask.delayed. It went from 60 seconds to 50 with the delayed call, so there seems to be some benefit with big files. Test included below

```python
import math

import cartopy.io.shapereader as shpreader
import numpy as np

import iris
import iris.tests as tests
import iris.quickplot as qplt
import matplotlib.pyplot as plt
from iris.util import mask_cube_from_shapefile


class TestCubeMasking(tests.IrisTest):
    """integration tests of mask_cube_from_shapefile
    using different projections in iris_test_data -
    values are the KGO calculated using ASCEND
    """

    def test_era5_russia(self):
        ne_countries = shpreader.natural_earth(
            resolution="10m", category="cultural", name="admin_0_countries"
        )
        self.reader = shpreader.Reader(ne_countries)
        cube = iris.load_cube(
            "/data/users/achamber/ERA5_data/era5_orography.nc"
        )
        ne_russia = [
            country.geometry
            for country in self.reader.records()
            if "Brazil" in country.attributes["NAME_LONG"]
        ][0]
        masked_test = mask_cube_from_shapefile(
            cube, ne_russia, minimum_weight=0.1
        )
        print(np.sum(masked_test.data))
        qplt.pcolormesh(masked_test)
        plt.savefig("test_big_c1.png")
        assert np.sum(masked_test.data) > 5000
```

Reply:
Trying it with the dask API docs recommended formatting is slower (61 seconds) than the current code (47 seconds). Why? Who knows.

Reply:
Thanks very much for the code for me to replicate. I put `start = time.time()` at the start of the test then put `assert((time.time() - start) == 0)` at the end so I could see the duration (hacky, but effective).

> It went from 60 seconds to 50 with the delayed call

I get 50 either way.

> Trying it with the dask API docs recommended formatting is slower (61 seconds) than the current code (47 seconds). Why? Who knows.

I would hazard this is because `dask.delayed` is adding more overhead than it is saving.